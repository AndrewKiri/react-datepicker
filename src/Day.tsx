import * as React from 'react';

interface Props{
  value: string,
  customClass: string
}

const Day: React.SFC<Props> = ({
  value,
  customClass
}) => {
  return (
    <td className={customClass}>
      {value}
    </td> 
  )
}

export default Day;