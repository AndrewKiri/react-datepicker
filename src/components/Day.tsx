import * as classNames from 'classnames';
import * as moment from 'moment';
import * as React from 'react';

interface Props {
  value: string;
  customClass?: string;
  customText?: string;
  onChange?: (value: string) => void;
  start?: boolean;
  end?: boolean;
  range?: boolean;
  today?: boolean;
  selected?: boolean;
}

class Day extends React.Component<Props> {
  public static defaultProps = {
    value: '',
  };

  public handleChange = () => {
    const { onChange, value } = this.props;
    // click event common process add here
    if (onChange) {
      onChange(value);
    }
  };

  public render() {
    const { value, customClass, customText, selected, start, end, range, today } = this.props;

    const dateClass = classNames('calendar__day', customClass, {
      'calendar__day--end': end,
      'calendar__day--range': range,
      'calendar__day--selected': selected,
      'calendar__day--start': start,
      'calendar__day--today': today,
    });

    return (
      <td onClick={this.handleChange} className={dateClass}>
        <span>{value}</span>
        {customText && <span className="calendar__day--text">{customText}</span>}
      </td>
    );
  }
}

export default Day;
