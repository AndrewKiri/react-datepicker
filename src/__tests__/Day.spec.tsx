import { shallow } from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';
import Day from '../Day';

describe('<Day/>', () => {
  it('renders the day with only value', () => {
    const component = shallow(<Day value="4" />);
    expect(component.contains('4')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('simulate click event', () => {
    const onChange = sinon.spy();
    const component = shallow(<Day onChange={onChange} />);

    component.simulate('click');
    expect(onChange).toHaveProperty('callCount', 1);
  });

  it('class start, end, today, selected, range', () => {
    const component = shallow(<Day start end today selected range />);

    expect(component.hasClass('calendar__day--start')).toBeTruthy();
    expect(component.hasClass('calendar__day--end')).toBeTruthy();
    expect(component.hasClass('calendar__day--today')).toBeTruthy();
    expect(component.hasClass('calendar__day--selected')).toBeTruthy();
    expect(component.hasClass('calendar__day--range')).toBeTruthy();
  });

  it('custom class correctly', () => {
    const component = shallow(<Day customClass="test" />);

    expect(component.hasClass('test')).toBeTruthy();
  });

  it('custom text correctly', () => {
    const component = shallow(<Day customText="start" />);

    expect(component.text()).toEqual('start');
  });
});
