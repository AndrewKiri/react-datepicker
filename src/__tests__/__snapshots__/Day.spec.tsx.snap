// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day/> renders the day with only value 1`] = `
<td
  className="calendar__day"
  onClick={[Function]}
>
  <span>
    4
  </span>
</td>
`;
